analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -433,19 +433,24 @@ class Pip(
     private fun getLicenseFromLicenseField(value: String?): String? =
         value?.let {
             // Work-around for projects that declare licenses in classifier-style syntax.
-            getLicenseFromClassifier(it) ?: it
+            // Some use just "OSI Approved :: " instead of "License :: OSI Approved :: ".
+            getLicenseFromClassifier(it, "OSI Approved :: ") ?: it
         }?.takeUnless {
             it.isBlank() || it == "UNKNOWN"
         }
 
-    private fun getLicenseFromClassifier(classifier: String): String? {
+    private fun getLicenseFromClassifier(classifier: String, extraPrefix: String? = null): String? {
         // Example license classifier:
         // "License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)"
-        return if (classifier.startsWith("License :: OSI Approved :: ")) {
-             classifier.substringAfterLast(" :: ", missingDelimiterValue = "").trim().ifEmpty { null }
-        } else {
-            null
+        val prefixes = listOfNotNull("License :: OSI Approved :: ", extraPrefix)
+
+        prefixes.forEach { prefix ->
+            if (classifier.startsWith(prefix)) {
+                return classifier.substringAfterLast(" :: ", missingDelimiterValue = "").trim().ifEmpty { null }
+            }
         }
+
+        return null
```

Comment:
Based on my previous suggestion, I'd extend it like

    private fun getLicenseFromClassifier(classifier: String): String? {
        // Example license classifier:
        // "License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)"
        val classifiers = classifier.split(" :: ").map { it.trim() }
        val licenseClassifiers = listOf("License", "OSI Approved")
        val license = classifiers.takeIf { it.first() in licenseClassifiers }?.last()
        return license?.takeUnless { it in licenseClassifiers }
    }

instead.

Reply:
Are you aware that the work-around should only be applied for the caller `getLicenseFromLicenseField()` but not for all other callers?

Reply:
I realize your current code is written like that, but actually I do not see any risk with making the code a bit simpler and apply the work-around in any case.

Reply:
Looked again at the classifiers and I agree there is not much risk, https://pypi.org/classifiers/.
Want me to squash so that we just have one commit ?

Reply:
I'd be fine with squashing.